yarpl/CMakeLists.txt

```diff
@@ -21,18 +21,28 @@ set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-weak-vtables -Wno-padded")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fno-omit-frame-pointer")
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -momit-leaf-frame-pointer")
 
+if(YARPL_WRAP_SHARED_IN_LOCK)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DYARPL_WRAP_SHARED_IN_LOCK")
+    message("Compiler lacks support std::atomic<std::shared_ptr>; wrapping with a mutex")
+elseif(YARPL_WRAP_SHARED_IN_ATOMIC)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DYARPL_WRAP_SHARED_IN_ATOMIC")
+    message("Compiler lacks std::shared_ptr atomic overloads; wrapping in std::atomic")
```

Comment:
What is the difference between this and the above? Both mention the same thing lacking but different workarounds?

Reply:
Newer compilers/libstdc++ have `std::atomic_{exchange,load,store}` specialized for `std::shared_ptr<T>` directly, so the `std::atomic` structure never shows up. If that's not supported, the code falls back to wrapping an `std::shared_ptr` in `std::atomic` structure. If *that* isn't supported either (my local clang doesn't like it), it falls back further to locking with a mutex and shimming `{exchange,load,store}`.